libs/thredds/odc/thredds/__init__.py

```diff
@@ -0,0 +1,77 @@
+"""Thredds crawling and YAML fetching utilities
+"""
+from thredds_crawler.crawl import Crawl
+import requests
+from urllib.parse import urlparse
+from multiprocessing.dummy import Pool as ThreadPool
+
+
+def thredds_find_glob(
+    base_catalog: str, skips: list, select: list, workers: int = 8
+) -> list:
+    """Glob YAML's from base Thredds Catalog recursively
+    
+    Arguments:
+        base_catalog {str} -- Base of the catlog to crawl from
+        user_skips {list} -- Paths to skip in addition to NCI specific defaults
+        select {list} -- Paths to select (useful YAML's)
+        workers {int} -- Number of workers to use for Thredds Crawling
+    
+    Returns:
+        list -- List of datasets to Index
+    """
+    user_skips = Crawl.SKIPS
+    user_skips = user_skips.extend(skips)
+
+    results = Crawl(
+        base_catalog + "/catalog.xml", select=select, skip=user_skips, workers=workers
+    ).datasets
```

Comment:
Is `results` a list of object or a stream? If its is a list then fine, but if it can be "lazy" then we should also return generator and not a list.

Reply:
It is a list , I would like to make is lazy and return iterators in the future. This is a refactor than a rewrite. Internally crawl uses multiprocessing.